prov/gni/src/gnix_vc.c

```diff
@@ -677,7 +678,6 @@ static int __gnix_vc_connect_to_self(struct gnix_vc *vc)
 	}
 
 	vc->peer_id = vc->vc_id;
-	vc->peer_caps = ep->caps;
 	vc->peer_irq_mem_hndl = ep->nic->irq_mem_hndl;
```

Comment:
I'm still concerned that conn_state is getting set before we finish filling in the vc struct leading to other issues with fields not filled in when we check for the connected state. I would rather we move the setting of conn_state to the last struct member we fill in here.